dali/python/backend_impl.cc

```diff
@@ -51,6 +51,8 @@ static std::string FormatStrFromType(TypeInfo type) {
     return py::format_descriptor<double>::format();
   } else if (IsType<bool>(type)) {
     return py::format_descriptor<bool>::format();
+  } else if (IsType<float16>(type)) {
+    return "f2";
```

Comment:
py::format_descriptor<T>::format() returns 1 char, while we return two here. Won't it break anything?

Reply:
dtype description in numpy is 1 char for the most common types, but it actually accepts string. "f2" means 2-byte float (which is float16)